pkg/status/controller.go

```diff
@@ -143,6 +202,83 @@ func (c *Controller) run(ctx context.Context, workers int, cListers chan interfa
+	_, err := c.statusCollectorInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			sc := obj.(*v1alpha1.StatusCollector)
+			logger.V(5).Info("Enqueuing reference to StatusCollector because of informer add event", "name", sc.Name, "resourceVersion", sc.ResourceVersion)
+			c.workqueue.Add(statusCollectorRef(sc.Name))
+		},
+		UpdateFunc: func(old, new interface{}) {
+			oldSc := old.(*v1alpha1.StatusCollector)
+			newSc := new.(*v1alpha1.StatusCollector)
+			if oldSc.Generation != newSc.Generation {
+				logger.V(5).Info("Enqueuing reference to StatusCollector because of informer update event", "name", newSc.Name, "resourceVersion", newSc.ResourceVersion)
+				c.workqueue.Add(statusCollectorRef(newSc.Name))
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			if typed, is := obj.(cache.DeletedFinalStateUnknown); is {
+				obj = typed.Obj
+			}
+			sc := obj.(*v1alpha1.StatusCollector)
+			logger.V(5).Info("Enqueuing reference to StatusCollector because of informer delete event", "name", sc.Name)
+			c.workqueue.Add(statusCollectorRef(sc.Name))
+		},
+	})
+	if err != nil {
+		c.logger.Error(err, "failed to add statuscollectors informer event handler")
+		return err
+	}
+	return nil
+}
+
+func (c *Controller) setupCombinedStatusInformer(ctx context.Context, ksInformerFactory ksinformers.SharedInformerFactory) error {
+	logger := klog.FromContext(ctx)
+	c.combinedStatusInformer = ksInformerFactory.Control().V1alpha1().CombinedStatuses().Informer()
+	c.combinedStatusLister = ksInformerFactory.Control().V1alpha1().CombinedStatuses().Lister()
+	_, err := c.combinedStatusInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			cs := obj.(*v1alpha1.CombinedStatus)
+			logger.V(5).Info("Enqueuing reference to CombinedStatus because of informer add event", "name", cs.Name, "resourceVersion", cs.ResourceVersion)
+			c.enqueueCombinedStatus(cs)
+		},
+		UpdateFunc: func(old, new interface{}) {
+			oldCs := old.(*v1alpha1.CombinedStatus)
+			newCs := new.(*v1alpha1.CombinedStatus)
+			if oldCs.Generation != newCs.Generation {
+				logger.V(5).Info("Enqueuing reference to CombinedStatus because of informer update event", "name", newCs.Name, "resourceVersion", newCs.ResourceVersion)
+				c.enqueueCombinedStatus(newCs)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			if typed, is := obj.(cache.DeletedFinalStateUnknown); is {
+				obj = typed.Obj
+			}
+			cs := obj.(*v1alpha1.CombinedStatus)
+			logger.V(5).Info("Enqueuing reference to CombinedStatus because of informer delete event", "name", cs.Name)
+			c.enqueueCombinedStatus(cs)
+		},
+	})
+	if err != nil {
+		c.logger.Error(err, "failed to add combinedstatuses informer event handler")
+		return err
+	}
+	return nil
+}
+
+func (c *Controller) enqueueCombinedStatus(obj interface{}) {
+	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
```

Comment:
FYI: `obj` is never a `cache.DeletedFinalStateUnknown` (look at the call sites). The code producing and consuming the work queue entries would be simpler if they were based on `cache.ObjectName` rather than an encoded string.